github-actions/npm/checkout-and-setup-node/action.yml

```diff
@@ -38,11 +32,11 @@ inputs:
 runs:
   using: composite
   steps:
-    - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+    - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
       with:
-        ref: ${{ inputs.ref }}
-        fetch-depth: ${{ inputs.fetch-depth }}
+        filter: tree:0
```

Comment:
do we also want the blobless filter? My understanding is that right now things like `git log` would result in the tree's being fetched- but also the blobs unnecessarily then. Filters can be combined as per git man

Reply:
On rereading it I think your right, but I am a bit fuzzy still.